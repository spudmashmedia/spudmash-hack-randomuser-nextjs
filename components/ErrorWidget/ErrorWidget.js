/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import React from "react";
import { Paper, ListItem, ListItemIcon, ListItemText } from "@material-ui/core";
import ErrorIcon from "@material-ui/icons/Error";

let ErrorWidget = props => {
  let errorMessage = `${props.message}`;
  return (
    <Paper>
      <ListItem>
        <ListItemIcon>
          <ErrorIcon color="error" />
        </ListItemIcon>
        <ListItemText primary={errorMessage} />
      </ListItem>
    </Paper>
  );
};

export default ErrorWidget;
