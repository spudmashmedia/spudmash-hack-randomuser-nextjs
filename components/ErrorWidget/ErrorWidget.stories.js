/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import React from "react";
import ErrorWidget from "./ErrorWidget";
import { withA11y } from "@storybook/addon-a11y";

export default {
  component: ErrorWidget,
  title: "ErrorWidget",
  decorators: [withA11y],
  excludeStories: /.*Data$/
};

export const actionData = "Something went wrong";

export const Default = () => <ErrorWidget message={actionData} />;
