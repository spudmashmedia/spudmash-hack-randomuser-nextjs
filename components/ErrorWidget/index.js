/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import ErrorWidget from "./ErrorWidget";
export default ErrorWidget;
