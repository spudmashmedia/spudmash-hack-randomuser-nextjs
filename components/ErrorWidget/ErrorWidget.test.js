import React from "react";
import ErrorWidget from "./ErrorWidget";
import renderer from "react-test-renderer";

describe("ErrorWidget Component Snapshot Test", () => {
  test("renders correctly", () => {
    const actionData = "Something went wrong..";

    const tree = renderer.create(<ErrorWidget message={actionData} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
