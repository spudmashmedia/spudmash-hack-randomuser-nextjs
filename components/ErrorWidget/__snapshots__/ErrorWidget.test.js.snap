// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorWidget Component Snapshot Test renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-colorError"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        Something went wrong..
      </span>
    </div>
  </li>
</div>
`;
